docs/getting_started.md

```diff
@@ -0,0 +1,144 @@
+# Getting started with SMART
+
+The basic SMART workflow is summarized in the diagram below. This is the general sequence of events followed in all of the SMART demos, which introduce applications of the software at increasing levels of complexity. Examples 1 and 2 give 2-dimensional cases, whereas examples 3-6 give increasingly complicated 3-dimensional cell signaling models.
+
+![SMART flow chart](images/SMART_flow_chart.png)
+
+To use SMART, the user should plan to write a python script (or Jupyter notebook) that goes through the outlined steps. The easiest way to get started with this is to follow the template of the SMART demo Jupyter notebooks or .py files. In the `smart` and `smart-lab` images, these files exist in the `/repo/examples` folder.
+
+Here, we walk through a minimal example demonstrating the use of SMART. In this simple example, the only required imports are `smart` and `dolfin`:
+```python
+import smart
+import dolfin
+```
+It is also helpful to define some units to be used in defining SMART objects. Here, we define a unit for the diffusion coefficient and for the concentration of a molecular species per unit surface area.
+```python
+D_unit = smart.units.unit.um**2 / smart.units.unit.s
+conc_unit = smart.units.unit.molecule / smart.units.unit.um**2
+```
+Following these preliminary steps, we begin the basic workflow of SMART:
+1. **Initialize SMART containers**: Each container (species, compartments, reactions, or parameters) is initialized by first creating individual objects of the class `smart.model_assembly.Species`, `smart.model_assembly.Compartment`, `smart.model_assembly.Reaction`, `smart.model_assembly.Parameter`. The initialization of these objects is described in some detail in Examples 1 and 3, or comprehensively in the API documentation.
+
+    These SMART containers contain all instances of any given class. In general, the user should define an empty container of the correct type and then add all instances of each object manually. In this simple example, we consider a model with a single species (`A`) over a single surface (`surf`), with a single reaction (`r1`, with parameter `k` defining the rate of decay) corresponding to exponential decay over time:
+    ```python
+    surf = smart.model_assembly.Compartment("surf", 2, smart.units.unit.um, 1)
+    A = smart.model_assembly.Species("A", 10, conc_unit, 1.0, D_unit, "surf")
+    k = smart.model_assembly.Parameter("k", 1, 1/smart.units.unit.s)
+    r1 = smart.model_assembly.Reaction("r1", ["A"], [], param_map={"k":"k"}, eqn_f_str="k*A")
```

Comment:
Agreed, do you think it is worth changing the parameter maps throughout the other examples for clarity, or is it sufficient to just have the clear example with explanation here? I've also thought we might consider allowing a user to bypass the parameter map if the parameter name in the equation matches the one given to the class object (but this could cause problems if the same name is used twice).

Reply:
I would only do it in this example, to illustrate the usage.

I don't think one should be able to bypass the parameter map (as it introduces alot of new possible errors)